### ISSUE TITLE
How to understand offset in torchvision.ops.deform_conv2d?

### ISSUE THREAD

**Issue by KingMV** [developer0hye/Simple-PyTorch-Deformable-Convolution-v2#15]
As we know,  the sampling grid of standard 3*3 conv  is {(-1,-1),...,(1,1)}. In the deformable conv with kernel size 3*3, the sampling grid is {(-1+p1_x,-1+p1_y),...,(1+p9_x,1+p9_y)}. For torchvision.ops.deform_conv2d,  each value in the offset map denotes {p1_x,....,p9_x;p1_y,.....p9_y}? Or, it is {(-1+p1_x),...,(1+p9_x);(-1+p1_y),...,(-1+p9_y)}? Look forward to your reply.

**Comment by developer0hye** [comment]
First one is right. However, the order is reversed.

p1_y,.....p9_y;p1_x,....,p9_x

**Comment by KingMV** [comment]
> First one is right. However, the order is reversed.
> 
> p1_y,.....p9_y;p1_x,....,p9_x

@developer0hye Thank you for your reply.  I observed that the standard sampling grid  {(-1,-1),...,(1,1)} is not considered in the offset_visualization.py. Why?
` 
offsets_y = offsets[:, :9]
offsets_x = offsets[:, 9:]
grid_y = np.arange(0, offset_tensor_h)
grid_x = np.arange(0, offset_tensor_w)
grid_x, grid_y = np.meshgrid(grid_x, grid_y)
sampling_y = grid_y + offsets_y.detach().cpu().numpy()
sampling_x = grid_x + offsets_x.detach().cpu().numpy()
sampling_y *= resize_factor_h
sampling_x *= resize_factor_w
 `

**Comment by developer0hye** [comment]
Because we already know where the standard conv sees!

**Comment by KingMV** [comment]
> Because we already know where the standard conv sees!

@developer0hye  Thank you, I got it. In fact, the sum of offset and its base coordinate (e.g. {(-1,-1)}) is the real sampling position of deformable Conv. To my eye,  your implementation only shows the offset change of deformable Conv.

**Comment by developer0hye** [comment]
@KingMV 

Yes, you are right! If you want to update this, feel free to contribute to this project!
